content/docs/rfcs/56/README.md

```diff
@@ -0,0 +1,412 @@
+
+Requests to leave, join, kick and ban, as well as key exchange messages, MUST be sent to the content topic derived from the public key of the community.
+The content topic MUST be the hex-encoded keccak-256 hash of the public key of the community.
+
+```
+contentTopic = hex(keccak256(publicKey))
+``` 
+
+## Community Management
+
+The flows for Community management are as described below.
+
+### Community Creation Flow
+
+1. The Community owner generates a public/private key pair.
+2. The Community owner configures the Community metadata, according to the wire format "CommunityDescription".
+3. The Community owner publishes the Community metadata on a content topic derived from the public key of the Community. 
+the Community metadata SHOULD be encrypted with the public key of the Community. <!-- TODO: Verify this-->
+The Community metadata MAY be sent during fixed intervals, to ensure that the Community metadata is available to members.
+The Community metadata SHOULD be sent every time the Community metadata is updated.
+4. The Community owner MAY advertise the Community out of band, by sharing the public key of the Community on other mediums of communication.
+
+### Community Join Flow (peer requests to join a Community)
+
+1. A peer generates a public/private key pair.
+2. The peer requests to join a Community by sending a "CommunityRequestToJoin" message to the Community.
+At this point, the peer MAY send a "CommunityCancelRequestToJoin" message to cancel the request.
+3. The Community owner MAY accept or reject the request.
+4. If the request is accepted, the Community owner sends a "CommunityRequestToJoinResponse" message to the peer.
+5. The Community owner then adds the member to the Community metadata, and publishes the updated Community metadata.
+
+### Community Join Flow (peer is invited to join a Community)
+
+1. The peer is invited to join a Community by the Community owner, by sending a "CommunityInvitation" message.
+2. A peer generates a public/private key pair.
+3. The peer decrypts the "CommunityInvitation" message, and verifies the signature.
+4. The peer requests to join a Community by sending a "CommunityRequestToJoin" message to the Community.
+5. The Community owner MAY accept or reject the request.
+6. If the request is accepted, the Community owner sends a "CommunityRequestToJoinResponse" message to the peer.
+7. The Community owner then adds the member to the Community metadata, and publishes the updated Community metadata.
+
+### Community Leave Flow
+
+1. A member requests to leave a Community by sending a "CommunityRequestToLeave" message to the Community.
+2. The Community owner MAY accept or reject the request.
+3. If the request is accepted, the Community owner removes the member from the Community metadata, and publishes the updated Community metadata.
+
+### Community Ban Flow
+
+1. The Community owner adds a member to the ban list, revokes their grants, and publishes the updated Community metadata.
```

Comment:
When an user is banned a re-key also happens if the community is encrypted. Should that be mentioned here?

Reply:
Addressed in b333c9c